### ISSUE TITLE
STAR 2.5.3a seg fault during genome indexing

### ISSUE THREAD

**Issue by grendon** [alexdobin/STAR#320]
Hi,

I tried several combinations of the following command and all of them abort with seg-fault.
I tried the command with and without the last three parameters.
I also tried these other values for memory on the compute node 40 50 even though I know that the genome is not big.

```
[biologin-1 ~]$ srun -p normal --mem=100GB -c 8 --pty bash
[compute-0-0 ]$ module load STAR/2.5.3a
[compute-0-0 ]$ STAR --runThreadN 8 --runMode genomeGenerate \
--genomeDir ./ --genomeFastaFiles Corma2_refgenome.fasta \
--sjdbGTFfile Corma2_GeneCatalog_genes_20150506.new.gtf \
--sjdbGTFtagExonParentTranscript transcript_id \
--sjdbGTFtagExonParentGene name --sjdbOverhang 99

Sep 07 12:54:44 ..... started STAR run
Sep 07 12:54:44 ... starting to generate Genome files
Sep 07 12:54:47 ... starting to sort Suffix Array. This may take a long time...
Sep 07 12:54:48 ... sorting Suffix Array chunks and saving them to disk...
Sep 07 12:54:55 ... loading chunks from disk, packing SA...
Sep 07 12:54:56 ... finished generating suffix array
Sep 07 12:54:56 ... generating Suffix Array index
Sep 07 12:55:09 ... completed Suffix Array index
Sep 07 12:55:09 ..... processing annotations GTF
Segmentation fault
```

The first few lines of the GTF file are these:

```
$ more Corma2_GeneCatalog_genes_20150506.new.gtf
##gff-version 2
##date 2017-09-06
scaffold_1      JGI     exon    978     2076    .       +       .       name "CE11_1287"; transcript_id "140";
scaffold_1      JGI     CDS     1105    1962    .       +       0       name "CE11_1287"; proteinId "12"; exonNumber "1"
scaffold_1      JGI     start_codon     1105    1107    .       +       0	name "CE11_1287";
scaffold_1      JGI     stop_codon      1960    1962    .       +       0	name "CE11_1287";
scaffold_1      JGI     exon    1694    2222    .       -       .       name "fgenesh1_kg.1_#_2_#_Locus6242v1rpkm12.24"; transcript_id "686891";
scaffold_1      JGI     CDS     2130    2222    .       -       0       name "fgenesh1_kg.1_#_2_#_Locus6242v1rpkm12.24"; proteinId "686763"; exonNumber "2"
scaffold_1      JGI     stop_codon      2130    2132    .       -       0	name "fgenesh1_kg.1_#_2_#_Locus6242v1rpkm12.24";
scaffold_1      JGI     exon    2359    3282    .       -       .       name "fgenesh1_kg.1_#_2_#_Locus6242v1rpkm12.24"; transcript_id "686891";
scaffold_1      JGI     CDS     2359    2886    .       -       0       name "fgenesh1_kg.1_#_2_#_Locus6242v1rpkm12.24"; proteinId "686763"; exonNumber "1"
scaffold_1      JGI     start_codon     2884    2886    .       -       0	name "fgenesh1_kg.1_#_2_#_Locus6242v1rpkm12.24";
scaffold_1      JGI     exon    4333    4970    .       -       .       name "CE23_13545"; transcript_id "152";

```

Attached is the Log.out file

Any help will be greatly appreciated!

**Comment by alexdobin** [comment]
Hi @grendon 

the GTF file has formatting problems: each "exon" line has to contain "transcript_id" and "gene_id". It looks like the "name" tag may be converted into the "gene_id", but you will need to check that.

Cheers
Alex

**Comment by grendon** [comment]
Thank you Alex. 

I know that the exon lines do not conform to defaults and that is why I added these params to the cmd

```
--sjdbGTFtagExonParentTranscript transcript_id 
--sjdbGTFtagExonParentGene
```

I think I found the cause of the problem.  The ref genome and the mitochondrial files contain fairly fragmented scaffolds. When I concatenated those two files, to create the 'official reference genome', the resulting file had some scaffolds with the same names.  

The solution was to rename all the scaffolds of the mitochondrial file; then concat them together; then rerun the STAR cmd. It worked!
